packages/perspective/src/js/perspective.js

```diff
@@ -1314,6 +1315,12 @@ module.exports = function(Module) {
             pdata = parse_data(data, cols, types);
         }
 
+        for (let i = 0; i < names.size(); i++) {
+            if (cols.indexOf(names.get(i)) === -1) {
+                pdata.types.splice(i, 1);
+            }
+        }
```

Comment:
I'm not sure this works the way you intend. As you splice values from the types array, the index value derived from the name array will no longer be valid.